src/functionalTest/java/uk/gov/hmcts/divorce/testutil/ServiceAuthenticationGenerator.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.hmcts.divorce.testutil;
+
+import io.restassured.RestAssured;
+import io.restassured.response.Response;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.test.context.TestPropertySource;
+
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.http.HttpHeaders.CONTENT_TYPE;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@TestPropertySource("classpath:application.yaml")
+@Service
+public class ServiceAuthenticationGenerator {
+
+    @Value("${s2s.name}")
+    private String s2sName;
+
+    @Value("${idam.s2s-auth.url}")
+    private String s2sUrl;
+
+    public String generate() {
+        return generate(this.s2sName);
+    }
+
+    public String generate(final String s2sName) {
+        final Response response = RestAssured
```

Comment:
there is one in https://github.com/hmcts/nfdiv-case-api/blob/master/src/functionalTest/java/uk/gov/hmcts/divorce/testutil/FunctionalTestSuite.java#L55 can we not use that ?

Reply:
I pinched this one from FunctionalTestSuite.java and the FunctionalTestSuite now uses this one, as I needed it to be more reusable across for other classes.